spacy/lang/nl/syntax_iterators.py

```diff
@@ -39,9 +39,14 @@ def noun_chunks(doclike: Union[Doc, Span]) -> Iterator[Tuple[int, int, int]]:
     # Label NP for the Span to identify it as Noun-Phrase
     span_label = doc.vocab.strings.add("NP")
 
+    start_span = -1
     # Only NOUNS and PRONOUNS matter
     for i, word in enumerate(filter(lambda x: x.pos in [PRON, NOUN], doclike)):
         # For NOUNS
+        # Prevent nested chunks from being produced
+        if word.left_edge.i <= start_span:
+            continue
+
```

Comment:
This is the right idea, but the `start_span` values that are used below aren't `word.left_edge.i`, so I think this check needs to be moved into each if block below with separate checks to compare to the `start_span` values that will be used by each alternative.

Reply:
I have tried this and don't think this actually works as it is still failing the test for noun_chunking. I believe that the noun_chunking code's logic actually has a problem. I'm trying to see if I can fix that.

Reply:
This should be possible without modifying anything else in the chunking algorithm. Can you commit what you've tried so far, I can take a look...

Reply:
I have pushed the commits. I basically tried all the combinations of moving the if statement in NOUN, PRON, and BOTH, and also tried to use end_span. 

Adding the if `word.left_edge.i` fixes the bug if it was added in NOUN portion but breaks the unit test `def test_chunking(nl_sample, nl_reference_chunking)` if you add it to PRON then the unit test keeps working, but it doesn't fix the bug.

Reply:
If you have pseudocode for Dutch noun chunking or with what labels does it happen because I think the usage of filtering is a bit off in this code but can't confirm it as I don't know Dutch.

Reply:
`word.left_edge.i` isn't the right thing to check because this code uses a different algorithm to choose the start token for the noun chunk spans.

You need to keep track of the end of the most recently yielded span and then check that the following start doesn't overlap with it, using the `start_span` in each if-block.

Let me add a test with the example from issue that checks whether overlapping spans have been produced.

Reply:
But isn't this what we are already doing? 
1- The `word.left_edge.i` gives us the following position of the first token of a phrase, which is the following start. 
2- We are keeping track of the end of the previous phrase in `end_span`. 
Why would we compare the first token with the start_span?

Reply:
No, in one case `start_span = min(children_i)` and in the other `start_span = word.i`. Neither of these are `word.left_edge.i`. See the proposal in #11275.

Reply:
It makes more sense after seeing your solution!

Reply:
Just to confirm I'll close this PR since your PR will be merged?

Reply:
Yes, let's close this in favor of the other PR. Thanks for bringing this to our attention again!